include/leveldb/iterator.h

```diff
@@ -64,6 +65,10 @@ class Iterator {
   // REQUIRES: !AtEnd() && !AtStart()
   virtual Slice value() const = 0;
 
+  // Riak specific:  if a database iterator, returns key meta data
+  // REQUIRES: Valid()
+  virtual KeyMetaData & keymetadata() const {return(keymetadata_); };
```

Comment:
I'm not suggesting you lose mutability of the `keymetadata_` member; I get why you did that. I was suggesting that the returned reference could also be const unless otherwise needed. I changed the declaration to

    virtual const KeyMetaData & keymetadata() const {return(keymetadata_); };

in my local copy and everything still builds and runs just fine. I'm just not sure if you expect the need to return a non-const reference to `keymetadata_` in the future.